### ISSUE TITLE
Primitivize: Enumerable can shadow custom mappings

### ISSUE THREAD

**Issue by plexus** [plexus/yaks#21]
It's currently not possible to define a custom Primitivize mapping for a class that implements Enumerable, the pre-set definition will take precedence.

**Comment by groyoh** [comment]
@plexus I took a look at this issue and I think that this could happen to any class that has an ancestor already mapped, right? Shouldn't it be fixed for these classes too?

**Comment by groyoh** [comment]
@plexus I took a look at this issue and I think that this could happen to any class that has an ancestor already mapped, right? Shouldn't it be fixed for these classes too?

**Comment by groyoh** [comment]
@plexus I took a look at this issue and I think that this could happen to any class that has an ancestor already mapped, right? Shouldn't it be fixed for these classes too?

**Comment by groyoh** [comment]
@plexus I took a look at this issue and I think that this could happen to any class that has an ancestor already mapped, right? Shouldn't it be fixed for these classes too?

**Comment by plexus** [comment]
Sure, but none of the default mappings are commonly subclassed (or at least shouldn't be, it's not considered good practice to subclass things like Hash or String). Enumerable on the other hand is very commonly mixed in to other classes, that's what it's for.

**Comment by groyoh** [comment]
Yeah sure, but I basically meant that a user may for example want to primitivize `BigNum` and `BigDecimal` differently, say using strings? Am I missing something here?

**Comment by plexus** [comment]
No, you're right. In that case you could still do it though by adding the subclass first and then the superclass. In the case of Enumerable it's simply not possible. (Except registering a new "empty" primitivizer, and defining all rules manually, with the least specific ones last).

So I guess in the case of classes/modules we should do a topographical sort.

**Comment by plexus** [comment]
s/topographical/topological/

**Comment by groyoh** [comment]
Ahah :laughing: I have a solution in mind, I'll come up with a PR and you'll tell me what you think.
